python/ray/train/tests/test_backend.py

```diff
@@ -236,6 +236,8 @@ def single_worker_fail():
     assert isinstance(exc.value.__cause__, ValueError)
 
 
+# TODO(@justinvyu: fix test and/or deprecate relevant code path)
+@pytest.mark.skip("Mocked execute_async doesn't work as intended")
```

Comment:
Is this intentional as part of this PR?

Reply:
yeah, paired with @justinvyu on this for some time, and we came to the conclusion that the mocking inside the test may need to be updated to be compatible with the fix in this PR, but we couldn't figure it out. I think @justinvyu said he can come back in the future to fix or remove the test, will also let him elaborate